rosbag2/src/rosbag2/sequential_reader.cpp

```diff
@@ -41,11 +45,13 @@ void
 SequentialReader::open(
   const StorageOptions & storage_options, const ConverterOptions & converter_options)
 {
-  storage_ = storage_factory_->open_read_only(storage_options.uri, storage_options.storage_id);
+  metadata_ = metadata_io_->read_metadata(storage_options.uri);
+  storage_ = storage_factory_->open_read_only(
+    metadata_.relative_file_paths[0], storage_options.storage_id);
```

Comment:
this should at least check if `relative_file_paths` is non empty otherwise this will lead to a segfault.